### DISCUSSION TITLE
[Note Template] Export and sync note with citekey as title

### DISCUSSION THREAD

**Discussion by bepolymathe** [windingwind/zotero-better-notes#618] [Note Templates]
Hi @windingwind 

Thanks for this plugin ! And happy new year. 
I happily use the syncronization of the notes between Zotero and Logseq (thanks to better-notes) but I would like the exported notes to be titled the number of "citekey" inherited from BetterBibtex. Is it possible ?

**Comment by windingwind** [comment]
Yes. Modify the ExportMDFilename template to fit your needs.

Let me know if you have any difficulties.

**Comment by bepolymathe** [comment]
Ok, i think it’s somethings with `<p>${topItem.citationKey ? topItem.citationKey : ""}</p>` 🤔 

in the ExportMDFilename... maybe like this ?

`${(noteItem.getNoteTitle ? noteItem.getNoteTitle().replace(/[/\\?%*:|"<> ]/g, "-") + "-" : "")}${topItem.citationKey }.md`

**Comment by windingwind** [comment]
The filename is a pure string, not an HTML element. Don’t wrap it with <p>.

Not sure if citation key can be get directly. I guess you should use getField?

**Comment by bepolymathe** [comment]
My dream is to have something like this as result

![image](https://user-images.githubusercontent.com/29939805/210390515-2dfca2d5-fd6e-41cb-9289-aa2a7497db1e.png)

**Comment by bepolymathe** [comment]
> I guess you should use getField?

I have to learn about this...

**Comment by windingwind** [comment]
Yes, it is possible. I’m not familiar with BBT thus not able to provide the template you beed right now. Tomorrow I will have time to test it.

Don’t worry about it. It’s easy.

**Comment by bepolymathe** [comment]
Thanks for your help, I'm not a developer. In my example below, the export should be formatted with metadata which is the logseq format (a little different from classic Yaml).

![image](https://user-images.githubusercontent.com/29939805/210392295-bcc8b58c-94f1-47e0-ac40-1703d6b73c0c.png)

**Comment by windingwind** [comment]
More details about the logseq yaml format would be helpful. Does logseq has any private format? Are there any general yaml features that logseq does not support?

**Comment by windingwind** [comment]
Sorry I misunderstood the md titles you mentioned here. The title refers to the first line of the exported md file, not the filename, right?

**Comment by bepolymathe** [comment]
I was thinking well of the file name and not the first line of the file. My idea is to use the MarkDBConnect plugin which detects files with the name of the BBT citkey and indicates it in Zotero...

![image](https://user-images.githubusercontent.com/29939805/210396710-cbc9febb-7056-4eda-8976-84b11e8ac720.png)

**Comment by windingwind** [comment]
Try this for your `ExportMDFileName`:

```
@${(noteItem.parentItem && noteItem.parentItem.getField("citationKey")) || noteItem.key}.md
```

The YAML header is defined in `ExportMDFileHeader`.

**Reply by aathanor** [reply]
For me this code does not get the citation key, The filename contains only the itemKey. If I remove the || noteItem.key, I get "@undefined". Tested it by selecting individual notes, group of notes or the parent item itself (the way I use it normally).

Funny thing, ExportMDFilenameV2 shows the correct preview in Template editor, with the citation key. But if saving using the template, with the same note selected, the result is @undefined.

**Comment by bepolymathe** [comment]
Thanks, it works for the file name. 

For the metadata export step, if I understand correctly, I absolutely must keep the YAML at the head so that Zotero Better Notes recognizes the note by his ID and is able to synchronize it 🤔 
 
The format recognized in Logseq for a good quality display would assume that I arrive at something like this as a header.

```
alias:: TITLE
tags:: [[tag1]]
type:: [[ZoteroType]]; 
source:: [[zotero]] 
date::  2022
pages:: 84
url:: [Web site](https://www...)
zotero_link:: [citekey](zotero://select/items/@citekey)
file:: ![TITLE](file:///home/bepolymathe/Zotero/storage/PQU6ZZ3C/PDF_NAME.pdf)
icon:: 📚

```

The question I ask myself. Could it be possible that the ID for Zotero Better Notes is included in the top file metadata in logseq ? By adding for example a line like this 

`BetterNotes:: noteItem.key`

**Comment by anandkumar89** [comment]
**looking for**: `@citeKey/noteTitle` in  `alias` to each `item-note` created in zotero. Instead of noteTitle, shortTitle can also be used when available.

**my use case** : cite keys are **easier to remember** and **aliases are searchable** in various note taking apps (e.g. obsidian [[@citekey...]] will suggest all notes with specific aliases)

Until now I used obsidian plugins to cite literature, but that would be rendered obsolete as zotero/betternotes is sufficient for all literature note taking.

**Comment by duduOliver** [comment]
The following code helps and hope it works for you:
`header.citationKey = noteItem.parentItem? noteItem.parentItem.getField("citekey") : "";`
